### DISCUSSION TITLE
Server Components and Server Actions with an external API - Is rate limiting a concern?

### DISCUSSION THREAD

**Discussion by dylanmooney** [vercel/next.js#64868] [Help]
### Summary

I'm building a Next.js app that integrates with an already existing backend API written in Laravel.

It has basic rate limitting configured for each endpoint at 60 requests per a minute.

Due to the frontend being on a server, given enough concurrent users make a request at the same time, will the frontend be rate-limited?

I'm aware Vercel use a dynamic range of IP's. However, I'm not sure if the addresses are different for each serverless function invocation or if it is poolws when they are called within a short timeframe.

I had considered sending the X-Forwarded-For header as part of request to the API, but there's some security implications involved there. I also don't have much say over this backend as it's worked on by another team within the company.

Am I better just utilising next.js for the initial render on the server and then fetching the data on the client using react-query?

**Comment by Kartik213** [comment]
@dylanmooney  hello, I am also facing this issue.
Did you find any solution for this?
